### PR TITLE
Add pre-commit for black and flake8

### PR DESCRIPTION
**Pull request by benjeffery**
Extracting pre-commit work from #471 for just black and flake8 so it can be merged before the rest.
Not yet ready for review as I'm getting the github action working.

### CODE REVIEW

**Comment by benjeffery** [comment]
@jeromekelleher As part of this PR I was going to move the pre-commit checks to a github action, but it seems I don't have permission to do this - the message is "Only repository collaborators may create workflows."

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: pre-commit
```
Comment (jeromekelleher) [inline .github/workflows/pre-commit.yml]: Maybe "lint" is a better name for this action? "pre-commit" is a bit confusing and just an implementation detail of how we're implementing the lint check.

File: docs/development.rst [hunk 2]
```diff
@@ -33,11 +33,13 @@ Quickstart
...
 - Run the tests to ensure everything has worked: ``python -m nose -vs``. These should
   all pass.
-
-- Make your changes in a local branch, and open a pull request on GitHub when you
-  are ready. Please make sure that (a) the tests pass before you open the PR; and
-  (b) your code passes PEP8 checks (see :ref:`sec_development_code_style` below
-  for a git commit hook to ensure this happens automatically) before opening the PR.
+- Install the pre-commit checks: ``pre-commit install``
+- Make your changes in a local branch, on each commit checks will be run for code style
+  and common problems (see :ref:`sec_development_code_style`), sometimes these will
```
Comment (jeromekelleher) [inline docs/development.rst]: This should be two sentences I think?

**Comment by jeromekelleher** [comment]
> @jeromekelleher As part of this PR I was going to move the pre-commit checks to a github action, but it seems I don't have permission to do this - the message is "Only repository collaborators may create workflows."

I don't really know how to fix this --- I've looked through the settings and can't find an easy way to add you as a collaborator. I had hoped to make you an admin of tskit-dev in a few weeks once you'd met the rest of the core developers (and then you'd have full admin rights for everything).

In the mean time, I guess we could merge this PR an I can follow up with whatever changes are needed to turn on the action?

File: docs/development.rst (hunk 2 above)
Comment (petrelharp) [reply docs/development.rst]: also a bit more explanation would be helpful for people unfamiliar - e.g. say that the code changes are formatting only; that the code is guaranteed to do the same thing.

Comment (benjeffery) [reply docs/development.rst]: Agreed, I've also added quite a bit more text about how to run checks without committing, and committing without checks.

**Comment by benjeffery** [comment]
In retrospect I think I am jumping the gun by adding the github action. I've reverted to using the current CI. Let's get this set of work wrapped up and schedule a CI/actions review for later?

**Comment by jeromekelleher** [comment]
OK, merged, thanks @benjeffery. FYI I've used the "rebase and merge" option here (the "merge" option isn't available now because we've selected "enforce linear history").

**Comment by benjeffery** [comment]
Looks like the only difference is that merging the PR didn't add a merge commit. Its a trade off of simpler history Vs recording each PR merge.

**Pull request merged**